### PR TITLE
test: Removes IDs From Resource Subaccount Service Plan Tests

### PR DESCRIPTION
**Pull request by andre-dossinger**
## Purpose
* Removes fixed IDs from tests to enable landscape independent fixture recording. 

## Does this introduce a breaking change?
```
[ ] Yes
[ x ] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[ ] Feature
[ x ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code via automated test

```bash
go test ./...
```

## What to Check

Verify that the following are valid:

* Automated tests are executed successfully

## Checklist for reviewer
The following organizational tasks must be completed before merging this PR:

* [x] The PR is assigned to the Terraform project and a status is set (typically "in review").
* [x] The PR has the matching labels assigned to it.
* [x] The PR has a milestone assigned to it.
* [x] If the PR closes an issue, the issue is referenced.
* [x] Possible follow-up items are created and linked.

### CODE REVIEW

**Comment by andre-dossinger** [comment]
Prerequisite for #59

File: internal/provider/datasource_subaccount_service_plan_test.go [hunk 1]
```diff
@@ -87,24 +89,29 @@ func TestDataSourceSubaccountServicePlan(t *testing.T) {
 
 }
 
-func hclDatasourceSubaccountPlanById(resourceName string, subaccountId string, planId string) string {
+func hclDatasourceSubaccountPlanByIdBySubaccountIdFromFilteredList(resourceName string, subaccountName string, planName string) string {
```
Comment (andre-dossinger) [inline internal/provider/datasource_subaccount_service_plan_test.go]: This function now selects any service plan with a matching name. For the purpose of the test this should be sufficient I think.

Comment (lechnerc77) [reply internal/provider/datasource_subaccount_service_plan_test.go]: I am afraid this won't do the trick. Service plan names are not unique, only in combination with the offering name the plan ID can be clearly identified.

**Review by lechnerc77 — CHANGES_REQUESTED** [review]
Following up your comment - the plan name standalone won't be sufficient to determine the plan_id

**Pull request merged**